rfcs/0020-security-on-call.md

```diff
@@ -0,0 +1,165 @@
+
+Our process for security issues is currently fairly well executed, but
+porly defined. How to participate and how to do the job is very
+nebulous and boils down to:
+
+1. Everybody pays attention
+2. Everybody patches
+
+This system means there is no division of labor, and no documented way
+to ensure patches are patched the same way, every time.
+
+## 2. Community and Commitment Sensitivity
+
+As a community, we have a few dedicated contributors patching
+the vast majority of security issues. This has worked well, and has
+even ensured fairly good coverage for some time now.
+
+Security patching work is an easy way to burn contributors out. The
+perpetual feed of new issues to fix is exhausting. This is concerning,
+because most of the patches are being applieed by highly skilled,
+involved, and "core" members of the NixOS community. It would be a
+shame to lose them.
+
+At the same time, most security patches are easy to apply:
+
+ - Most announcements to security email lists include easy-to-apply
+   patches.
+ - Well established, well funded distributions have full time
+   employees focusing on security matters. These other distributions
+   regularly publish their minimal security patches on their bug
+   trackers or in their source trees.
+ - Many security issues can be easily fixed by minor package bumps.
+
+Some of the more tricky parts are determining how to backport the
+patch, and finding build capacity for testing large rebuild changes.
+Both of these questions can easily be answered by asking more
+experienced community members for help.
+
+# Detailed design
+[design]: #detailed-design
+
+I propose we create an on-call rotation for publicly disclosed
+security issues. Each member of the team will be on call for 24 hours,
+and expected to handle every new issue which is created within their
+on-call period.
+
+The patching team will not handle issues under embargo.
+
+## Patching Team
+
+The team should be of at least 10 members, preferably over 14.
+
+Members of this team should range from new contributors looking to
+participate, to more skilled and well known contributors.
+
+### Requirements
+
+Each team member should:
+
+ - Know how patches work, or at least be willing to learn
+ - Be comfortable with Git and our backporting workflow
+ - Know their personal limits and be confident asking for help
+
+## On Call
+
+An on-call rotation system should be used or made to handle scheduling
+and informing people about their shift. A shift is 24hrs long, and
+should probably start at midnight UTC, to be equally unfair to
+everyone.
+
+### Responsibilities
+
+1. Monitor a well defined list of mailing lists for new issues.
+2. Ensure each issue is triaged and addressed if needed.
+
+#2 is a bit vaguely worded, as the person is not required to
+_actually_ fix the issue. They are allowed to delegate the patching to
+other people. However, they _are_ responsible for ensuring the issue
+is _fixed_.
```

Comment:
Once we have a MAINTAINERS file, we could delegate security issues in non-critical packages (let's say not in `nixos-small`) to the respective maintainers by default.